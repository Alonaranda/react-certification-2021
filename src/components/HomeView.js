import React, { useEffect, useState } from 'react'
import { HeaderNav } from './HeaderNav';
import { HomeScreen } from './HomeScreen';



export const HomeView = ({ defaultCategorias = [] }) => {


    const [categorias, setCategorias] = useState(defaultCategorias);
    useEffect(() => {
        categorias.shift();
    }, [categorias])


    return (
        <div>
            <HeaderNav setCategorias={setCategorias} />
            <ol>
                {
                    categorias.map(categoria =>
                        <HomeScreen
                            key={categoria}
                            categoria={categoria}
                        />
                    )
                }
            </ol>

        </div>
    )
}
