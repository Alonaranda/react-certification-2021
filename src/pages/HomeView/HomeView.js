import React from 'react';
import { Link } from 'react-router-dom';

export const HomeView = () => {
  return (
    <>
      <h1>Aqui arriba voy a poner un Header</h1>

      <p>Aqui voy a poner vidios</p>

      <pre>
        <Link to="/"> ← go back</Link>
      </pre>
    </>
  );
};
