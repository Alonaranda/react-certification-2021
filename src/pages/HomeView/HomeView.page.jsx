import React from 'react';
import { Link } from 'react-router-dom';
import Header from '../Header/Header';
import Fill from '../Fill/Fill'

const HomeView = () => {



  return (
    <div>
      <Header/>

      <Fill/>

      <pre>
        <Link to="/"> ← go back</Link>
      </pre>
    </div>
  );
};

export default HomeView;
