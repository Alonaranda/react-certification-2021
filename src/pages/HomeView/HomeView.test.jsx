import React from 'react'
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import HomeView from "./HomeView.page";

describe('Test HomeView', () => {
    it('should render', () => {
        const wrapper = shallow(<HomeView/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should have header', () => {
        const wrapper = shallow(<HomeView/>);
        const header = wrapper.find('Header');
        expect(header.length).toEqual(1);
        
    });

    it('should have fill', () => {
        const wrapper = shallow(<HomeView/>);
        const fill = wrapper.find('Fill');
        expect(fill.length).toEqual(1);
    });
});