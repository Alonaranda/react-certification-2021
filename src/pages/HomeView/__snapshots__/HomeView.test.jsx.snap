// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HomeView should look good 1`] = `ShallowWrapper {}`;
